cmake/kokkos_arch.cmake

```diff
@@ -459,7 +463,7 @@ IF (CMAKE_CXX_COMPILER_ID STREQUAL Clang AND WIN32)
 ENDIF()
 
 # MSVC ABI has many deprecation warnings, so ignore them
-IF (CMAKE_CXX_COMPILER_ID STREQUAL MSVC OR "x${CMAKE_CXX_SIMULATE_ID}" STREQUAL "xMSVC")
+IF (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC" OR "x${CMAKE_CXX_SIMULATE_ID}" STREQUAL "xMSVC")
```

Comment:
`MSVC` is a variable in `CMake`, see https://cmake.org/cmake/help/latest/variable/MSVC.html. Thus, we have to use quotes here.